src/components/flyout/deviceReconfigureFlyout.js

```diff
@@ -70,7 +70,7 @@ class DeviceReconfigureFlyout extends React.Component {
           .fromPromise(ApiService.getJobStatus(jobId))
           // Get completed jobs
           .filter(({ status }) => status === 3)
-          .flatMap(_ => deviceIds)
+          .flatMap(_ => deviceIds.filter(deviceId => devices.some(({ Id }) => deviceId === Id)))
```

Comment:
1) I think you should process the list of devices when it updates to create a helper data structure like a set of ids. Check for membership in the set instead of searching, potentially, the entire devices list multiple times.

2) Don't do the filter inside the flat map. Do it as part of the observable stream.
```
.flatMap(({ jobId, deviceIds }) =>
   ...
   .flatMap(_ => deviceIds)
)
.distinct()
.filter(deviceId => deviceIds.has(deviceId))
```